### ISSUE TITLE
Getting 403 when calling via Python

### ISSUE THREAD

**Issue by Julia288** [DaWe35/Similarweb-free-API#1]
Hey!
Thanks a lot for your contribution.
I am getting 403 when calling the URL via python, but it works when using the browser.
Do you have any idea how to deal with this?
Best, Julia

**Comment by pokrpokr** [comment]
Yeah, I got the same issue with Julia. Did the api ask for a free api-key? Anyway, thank you for offering this free api example!

**Comment by infinitum11** [comment]
> Yeah, I got the same issue with Julia. Did the api ask for a free api-key? Anyway, thank you for offering this free api example!

The only way to make this work is to send a request via Selenium / Puppeteer. Only a real browser can get a response. Other solutions like CLI cURL, PHP cURL, etc. won't work.

**Comment by Bender250** [comment]
It is enough to change the headers when you call it from python ;).

**Comment by Muhammadhassan1226** [comment]
In my Case, https://curlconverter.com/python/ works for me, as I copy the curl from the browser's network tab and paste it here. It provides me with Python-based code that works for me and has resolved the 403 Error.

<img width="1852" height="805" alt="Image" src="https://github.com/user-attachments/assets/fa88f6d1-1a17-49a3-abe2-e54ba83a2b48" />

**Comment by BelNaruto** [comment]
Hi, it looks like this issue has returned. I'm still getting a 403 error in the browser. Has anyone found a solution yet?

**Comment by Muhammadhassan1226** [comment]
@BelNaruto  Can you try my working trick ?

**Comment by BelNaruto** [comment]
Still the same error mate... It has changed, you can give it a try with your working trick

**Comment by simplesapien** [comment]
@Muhammadhassan1226 @BelNaruto @infinitum11 @Julia288 @pokrpokr 

If you've got the extension installed:

Right click icon -> Manage extension -> Toggle on 'Developer mode' -> Click on service worker -> Go to the network tab -> Open up any web page + click on the Similarweb extension to run it -> Find the request with 'similarweb' in the URL -> Use those request headers in your request.

I assumed that the cookie would've eventually limited me at 20 or 25 requests or whatever it is in the extension, but I've run it a ton of times with a lot of different URLs with no problems so far. If that does become an issue, the https://data.similarweb.com/api/v1/identity request could probably be used to refresh your cookie? idk.

**Comment by Bender250** [comment]
The extension was also dead two weeks when I noticed. I have reported the issue with the extension to SimilarWeb via their feedback form, no idea if they are reading it.

Once the extension works, it should be reasonable to reverse-engineer it again.

If you are researchers depending on SimilarWeb, you might want to check alternatives: https://measuretheweb.org/design/website_classification
